### PR TITLE
add in negative scaling find one flood workload

### PR DESCRIPTION
**Pull request by MAhmadShah**
**Jira Ticket:** [PERF-4749](https://jira.mongodb.org/browse/PERF-4749)

**Whats Changed:**  
We add in a new workload that is intended to test for negative scaling. This was originally created to replicate behaviour reported in HELP-44821.

**Patch testing results:**  
[Patch results](https://spruce.mongodb.com/version/64ef9419d1fe07c226429dfc/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) showing workload running on atlas-like instances

### CODE REVIEW

File: src/workloads/scale/NegativeScalingLoadStress.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            OperationCommand:
+              Filter: {_id: {^RandomInt: {min: 0, max: 100000}}}
+
+Actors:
+
+# Insert Actors
+- ActorFromTemplate:
+    TemplateName: InsertTemplate
+    TemplateParameters:
+      Name: Insert_1024
```
Comment (jawwadasghar) [inline src/workloads/scale/NegativeScalingLoadStress.yml:104]: Should this _1024 within the name be updated? 
I think it is usually either the number of docs or threads

File: src/workloads/scale/NegativeScalingLoadStress.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      Threads: 128
+      OnlyActiveInPhase: 1
+
+# Find Actors
+- ActorFromTemplate:
+    TemplateName: FindTemplate
+    TemplateParameters:
+      Name: Find_1024
```
Comment (jawwadasghar) [inline src/workloads/scale/NegativeScalingLoadStress.yml:112]: Same question here with the _1024

File: src/workloads/scale/NegativeScalingLoadStress.yml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+          - OperationName: insertOne
+            OperationCommand:
+              Document: *doc
+
+- TemplateName: FindTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "Find"}}
+    Type: CrudActor
+    Database: *dbname
+    ClientName: Default
```
Comment (jawwadasghar) [inline src/workloads/scale/NegativeScalingLoadStress.yml:83]: nit: update "Default" to something like "Query" or "Find"?

**Review by jawwadasghar — APPROVED** [review]
LGTM
minor nits about naming

**Comment by MAhmadShah** [comment]
@jawwadasghar Thank you for the comments, I've made all the updates!

File: src/workloads/scale/NegativeScalingLoadStress.yml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+              Filter: {_id: {^RandomInt: {min: 0, max: 100000}}}
+
+Actors:
+
+# Insert Actors
+- ActorFromTemplate:
+    TemplateName: InsertTemplate
+    TemplateParameters:
+      Name: Insert_128
+      Threads: 128
```
Comment (jimoleary) [inline src/workloads/scale/NegativeScalingLoadStress.yml:105]: suggestion: you could make this the default and lose one line.

File: src/workloads/scale/NegativeScalingLoadStress.yml [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      OnlyActiveInPhase: 1
+
+# Find Actors
+- ActorFromTemplate:
+    TemplateName: FindTemplate
+    TemplateParameters:
+      Name: Find_256
+      Threads: 256
```
Comment (jimoleary) [inline src/workloads/scale/NegativeScalingLoadStress.yml:113]: suggestion: you could make this the default and lose one line.

**Review by jimoleary — CHANGES_REQUESTED** [review]
Can we add the important metrics (I know it is pretty obvious here but it's still a good habit to get into).

**Comment by MAhmadShah** [comment]
> Can we add the important metrics (I know it is pretty obvious here but it's still a good habit to get into).

I've made the requested changes, ty Jim!

**Review by jimoleary — APPROVED** [review]
